### PR TITLE
@l2succes => Mark message read mutation

### PR DESCRIPTION
**Pull request by mzikherman**
This works with my WIP Emission branch for this!

I'll clean up/add specs shortly...

### CODE REVIEW

File: lib/loaders/loaders_with_authentication/impulse.js [hunk 1]
```diff
@@ -30,5 +32,13 @@ export default (accessToken, userID) => {
       },
       { method: "POST" }
     ),
+    markMessageReadLoader: impulseLoader(
```
Comment (mzikherman) [inline lib/loaders/loaders_with_authentication/impulse.js]: In our `event` framework for Impulse/Radiation, marking a message as read means logging an 'open' event for the relevant delivery.

File: schema/me/conversation/mark_read_message_mutation.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { GraphQLString, GraphQLNonNull } from "graphql"
```
Comment (mzikherman) [inline schema/me/conversation/mark_read_message_mutation.js]: This is a pretty standard mutation.

File: schema/me/conversation/index.js [hunk 3]
```diff
@@ -214,6 +214,33 @@ export const ConversationFields = {
     },
   },
 
+  // If the user is a recipient of the last message, return the relevant delivery id.
+  // This can be used to mark the message as read, or log other events.
+  last_message_delivery_id: {
```
Comment (mzikherman) [inline schema/me/conversation/index.js]: We need to grab this in the `Conversation` container fragment. Alternatively, we can bake this in to the mutation code (below).

Essentially, we need to know which 'delivery' to record the 'open' event for. Thus, at the `Conversation` level, we need to make available that relevant id. This is analogous to what we do further up in this file for `is_last_message_to_user`, `last_message_open`, etc. Indeed, they can probably share some code/get refactored. Probably not in this PR tho!

File: lib/loaders/loaders_with_authentication/impulse.js [hunk 4]
```diff
@@ -30,5 +32,13 @@ export default (accessToken, userID) => {
...
+      "events",
+      {
+        event: "open",
+        timestamp: moment().unix(),
```
Comment (mzikherman) [inline lib/loaders/loaders_with_authentication/impulse.js]: We generally send back unix timestamps (integers) to API's expecting date/time fields.

**Comment by mzikherman** [comment]
I think this is mergeable.

**Comment by alloy** [comment]
@mzikherman :ceiling-cat: remove the [WIP] part from the subject ;)

File: schema/me/conversation/index.js (hunk 3 above)
Comment (l2succes) [reply schema/me/conversation/index.js]: Ok, that makes sense. Yes, we should refactor the code for those fields as it's very similar in a future PR.

**Pull request merged**